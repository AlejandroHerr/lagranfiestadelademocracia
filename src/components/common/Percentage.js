import React from 'react';
import PropTypes from 'prop-types';

import { formatPercentage } from '../../utils/format';

const Percentage = ({ children, renderComponent: RenderComponent, ...props }) => (
  <RenderComponent {...props}>{`${formatPercentage(children).toString()}%`}</RenderComponent>
);

Percentage.propTypes = {
  children: PropTypes.number.isRequired,
  renderComponent: PropTypes.func.isRequired,
};

export default Percentage;
