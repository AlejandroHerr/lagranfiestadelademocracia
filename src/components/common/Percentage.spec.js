import React from 'react';
import renderer from 'react-test-renderer';

import Percentage from './Percentage';

describe('components/component/Percentage', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Percentage renderComponent={props => <div {...props} />}>{0.001}</Percentage>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
