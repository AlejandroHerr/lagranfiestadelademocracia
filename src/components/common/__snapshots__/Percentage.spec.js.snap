// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/component/Percentage renders correctly 1`] = `
<div>
  0.1%
</div>
`;
